p2panda-rs/src/schema/schema_description.rs

```diff
@@ -0,0 +1,73 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+use std::fmt;
+use std::fmt::Display;
+use std::str::FromStr;
+
+use crate::schema::error::SchemaDescriptionError;
+use crate::schema::validate::validate_description;
+
+/// The description of a schema which adheres to specification requirements. Used in the
+/// construction of `Schema`.
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub struct SchemaDescription(String);
+
+impl SchemaDescription {
+    /// Construct and validate new schema description from a string.
+    pub fn new(description: &str) -> Result<Self, SchemaDescriptionError> {
+        let description = Self(description.to_owned());
+        description.validate()?;
+        Ok(description)
+    }
+
+    /// Perform validation on the description string.
+    ///
+    /// 1. It consists of unicode characters
+    /// 2. ... and must be at most 256 characters long
+    pub fn validate(&self) -> Result<(), SchemaDescriptionError> {
+        if !validate_description(&self.0) {
+            return Err(SchemaDescriptionError::TooLongSchemaDescription);
+        }
+        Ok(())
+    }
+}
```

Comment:
Is there a reason you didn't use the validation trait here (and in other places)?

Reply:
No! I totally forgot about it 😱 I'll use it 👍